pkg/migration/azure_crs.go

```diff
@@ -391,6 +396,158 @@ func (m *azureMigrator) readRelease(ctx context.Context, ver string) error {
 	return nil
 }
 
+func (m *azureMigrator) updateCluster(ctx context.Context) error {
+	cluster := m.crs.cluster
+
+	// Drop operator version label.
+	delete(cluster.Labels, label.AzureOperatorVersion)
+
+	// Drop finalizers.
+	cluster.Finalizers = nil
+
+	// Adjust k8s apiserver bind port to match kubeadm.
+	if cluster.Spec.ClusterNetwork != nil && cluster.Spec.ClusterNetwork.APIServerPort != nil {
+		cluster.Spec.ClusterNetwork.APIServerPort = to.Int32Ptr(6443)
```

Comment:
Shouldn't that be set even if `cluster.Spec.ClusterNetwork.APIServerPort` is nil?

Reply:
It's optional. CAPZ defaults to `6443`: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b5ad5c4c8156cd02eb5447e4405c412c1b9dc64b/azure/scope/cluster.go#L461